jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/StepsRunner.java

```diff
@@ -332,17 +333,21 @@ private void pushBaseImageLayers() {
 
     results.baseImageLayerPushResults =
         executorService.submit(
-            () ->
-                scheduleCallables(
-                    PushLayerStep.makeList(
-                        buildContext,
-                        childProgressDispatcherFactory,
-                        results.targetRegistryClient.get(),
-                        Verify.verifyNotNull(
-                            results
-                                .baseImagesAndLayers
-                                .get()
-                                .get(results.baseImagesAndRegistryClient.get().images.get(0))))));
+            () -> {
+              List<List<Future<BlobDescriptor>>> baseImageLayerPushResults = new ArrayList<>();
+              for (List<Future<PreparedLayer>> baseImageLayers :
+                  results.baseImagesAndLayers.get().values()) {
+                List<Future<BlobDescriptor>> baseImageLayerPushResult =
```

Comment:
In a local scope, I think we don't always have to be fully verbose. IMO, reasonable names help reducing cognitive load when reading code, especially in a local scope. So, I prefer
```java
              List<List<Future<BlobDescriptor>>>  pushResults = new ArrayList<>();
              for (List<Future<PreparedLayer>> layers : results.baseImagesAndLayers.get().values()) {
                List<Future<BlobDescriptor>> pushResult =
```